tests/helpers/test_init.py

```diff
@@ -15,7 +15,7 @@ def test_extract_domain_configs() -> None:
         "zone 100": None,
     }
 
-    assert {"zone", "zone Hallo", "zone 100"} == set(
+    assert {"zone", "zone ", "zone Hallo", "zone 100"} == set(
```

Comment:
The behavior with partition is a bit different from the regex, I don't think that matters though?

Reply:
Is `"zone ": None` valid?

Reply:
Doesn't the regex make sure that there's a character after the space if there's a space after the domain?

Reply:
The point of this PR is to align with these cases which already use partition instead of the regex to make the code more readable and the behavior consistent

https://github.com/home-assistant/core/blob/11871cf9a57c6c5e6fefbc664516ebaf48ec56c6/homeassistant/bootstrap.py#L474
https://github.com/home-assistant/core/blob/11871cf9a57c6c5e6fefbc664516ebaf48ec56c6/homeassistant/config.py#L763
https://github.com/home-assistant/core/blob/11871cf9a57c6c5e6fefbc664516ebaf48ec56c6/homeassistant/helpers/check_config.py#L163

> Doesn't the regex make sure that there's a character after the space if there's a space after the domain?

Right, that's the difference. But I don't think it matters, except that it's a good idea to be consistent.

I can't find that splitting the configuration by "labels" is explained or documented anywhere except briefly mentioned here https://www.home-assistant.io/docs/configuration/splitting_configuration/, where it's referred to as "labels".